bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/finders/MDNSAddonSuggestionFinder.java

```diff
@@ -0,0 +1,125 @@
+import javax.jmdns.ServiceListener;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.io.transport.mdns.MDNSClient;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link MDNSAddonSuggestionFinder} for finding suggested Addons via MDNS.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonSuggestionFinder.class, name = MDNSAddonSuggestionFinder.SERVICE_NAME)
+public class MDNSAddonSuggestionFinder extends BaseAddonSuggestionFinder implements ServiceListener {
+
+    public static final String SERVICE_TYPE = "mdns";
+    public static final String SERVICE_NAME = SERVICE_TYPE + ADDON_SUGGESTION_FINDER;
+
+    private static final String NAME = "name";
+
+    private final Logger logger = LoggerFactory.getLogger(MDNSAddonSuggestionFinder.class);
+    private final Map<String, ServiceInfo> services = new ConcurrentHashMap<>();
+    private final MDNSClient mdnsClient;
+
+    @Activate
+    public MDNSAddonSuggestionFinder(@Reference MDNSClient mdnsClient) {
+        this.mdnsClient = mdnsClient;
+    }
+
+    public void addService(@Nullable ServiceInfo service) {
+        if (service != null) {
+            String qualifiedName = service.getQualifiedName();
+            if (services.put(qualifiedName, service) == null && logger.isTraceEnabled()) {
+                logger.trace("mDNS service name={}, properties={}", qualifiedName,
+                        Collections.list(service.getPropertyNames()).stream()
+                                .map(n -> n + "=" + service.getPropertyString(n)).toList());
+            }
+        }
+    }
+
+    @Deactivate
+    public void close() {
+        services.clear();
+        super.close();
+    }
+
+    @Override
+    public Set<AddonInfo> getSuggestedAddons() {
+        Set<AddonInfo> result = new HashSet<>();
+        addonCandidates.forEach(candidate -> {
+            candidate.getDiscoveryMethods().stream().filter(method -> SERVICE_TYPE.equals(method.getServiceType()))
+                    .forEach(method -> {
+                        Map<String, Pattern> map = method.getMatchProperties().stream().collect(
+                                Collectors.toMap(property -> property.getName(), property -> property.getPattern()));
+                        services.values().stream().forEach(service -> {
+                            if (method.getMdnsServiceType().equals(service.getType())
+                                    && propertyMatches(map, NAME, service.getName())
+                                    && Collections.list(service.getPropertyNames()).stream().allMatch(
```

Comment:
A match property defined in a candidate, but not defined in the service will not be tested here.